### PR TITLE
Scuba tanks for underwater diving.

### PR DESCRIPTION
**Pull request by nexusmrsep**
#### Summary
```SUMMARY: Features "Scuba tanks for diving underwater."```

#### Purpose of change
Divers use ~~oxygen~~ scuba tanks, but CDDA had only rebreathers and oxygenator, alternatively gills mutation to keep you up on oxygen.

#### Describe the solution
~~Added **automatic** use of oxygen tank and emergency oxygen tanks charges to breathe oxygen underwater, when running out of breath.~~

Added scuba tanks:
- 12L high pressure (232 bar) tank estimated at 60 minutes of diving (assumed shallow dive up to 10 meters deep)
- 4L high pressure (200 bar) reserve tank estimated at 20 minutes of diving (as above)

Tank needs to be worn and activated to work, and then is used automatically to replenish air needed to breathe underwater. It also protects from the effects of external gases like smoke, tear gas, toxic gas, as long as its worn and active.

Since the assumption here is that it is equipped with a on-demand regulator (that supplies air only when needed) it auto-turns-off when dropped.

Both tanks are quite hefty, so are rather designed to be worn (on straps) rather then carried freely around.

#### Describe alternatives you've considered
~~Considered manual use for that purpose, but went for quality of life. There is some extra risk behind it however, as tanks are not diving gear per se, and consume moves per use as defined in their use function.~~

Diving gas consumption vary with depth but since we don't have any underwater depths to visit (yet), this feature was not coded in.

#### Additional context
I've used this [CALCULATOR ](http://sb.fluomedia.org/diveplan/)to balance air consumption of the tanks, and few other sources on diving equipment.

### CODE REVIEW

**Comment by Rivet-the-Zombie** [comment]
Those are medical oxygen tanks - you cannot use that type of mask in place of SCUBA gear. It is just noy possible.

**Comment by nexusmrsep** [comment]
And that is why it takes 500 moves to use it as a cost. In compare it to "REBREATHER" flagged masks that use no moves to use. My logic here is - it's not designed for divers, but it's oxygen source nevertheless, so you can use it in a pinch.

If that logic is not valid - there are two options:
- scrapping this PR
- adding a diving mask that would be required as a tool used with oxy-tank (perhaps there is something similar already that can be repurposed?)

**Comment by ifreund** [comment]
Pretty sure medical oxygen tanks are pure O2 while dive tanks are mixed with nitrogen and helium.  You really couldn't use this as your only air supply even in a pinch as far as I know.

**Comment by nexusmrsep** [comment]
I've read [WIKI](https://en.wikipedia.org/wiki/Oxygen_toxicity) on oxygen toxicity, and it seems that a short exposure would be fine, prolonged might be very dangerous. So I have two propositions:

1. Pure medical oxygen `use_action` gives you stimulant effect, that in my tests seems to accumulate quite fast (but not lethal fast) under prolonged exposure (crossing wide deep river). I might bump up this effect and/or add other negative effects, including creating new `oxygen_toxicity` effect from scratch.
2. Add air tanks to the game with breathable mixture of O2 and N2 with proper diving masks, and place them in sporting goods shops etc.

Second option can be done regardless of whether option 1 passes or will be scrapped.
Please advise.

**Comment by Zireael07** [comment]
Option 2 would be cool for a second PR, methinks.

An oxygen toxicity effect would be good from a realism perspective, not too sure about gameplay implications.

**Comment by I-am-Erk** [comment]
Breathing pure O2 underwater is pretty problematic. I'm about to add nitrogen tanks (coded them in yesterday in fact), so it might be better to make a recipe for dive air rather then use straight oxygen. 

Also, the kind of rebreather mask on an oxygen tank would probably not work well underwater. Regulators are quite different.

**Comment by nexusmrsep** [comment]
Point taken. I'll rewrite this from scratch.

**Comment by nexusmrsep** [comment]
Reopened after rework. PR description updated.

File: data/json/items/tools.json [hunk 1]
```diff
@@ -5461,7 +5461,7 @@
     "id": "oxygen_tank",
     "type": "TOOL",
     "name": "oxygen tank",
-    "description": "This is tank of compressed medical oxygen with a regulator and mask.  Commonly used in emergency situations, it provides immediate relief for asthma attacks or smoke inhalation, and can provide a brief burst of energy.",
+    "description": "This is tank of compressed medical oxygen with a regulator and mask.  Commonly used in emergency situations, it provides immediate relief for asthma attacks or smoke inhalation, and can provide a brief burst of energy. Can also provide air underwater.",
```
Comment (Rivet-the-Zombie) [inline data/json/items/tools.json]: Both of the medical tank descriptions still need to be fixed.

Comment (nexusmrsep) [reply data/json/items/tools.json]: good catch, it slipped my mind

**Pull request merged**